pyro/infer/autoguide/guides.py

```diff
@@ -545,7 +545,9 @@ def get_posterior(self, *args, **kwargs):
         """
         Returns a MultivariateNormal posterior distribution.
         """
-        return dist.MultivariateNormal(self.loc, scale_tril=self.scale_tril)
+        transform = dist.transforms.LowerCholeskyAffine(self.loc, scale_tril=self.scale_tril)
```

Comment:
Please measure memory and time overhead introduced by this refactoring. This is a heavily used autoguide, and should be kept simple and fast. I am even inclined to introduce a new method `.get_posterior_transformed_dist()` or something so as not to obfuscate this code.

Reply:
I think we should add `get_base_dist` and `get_transform` methods, and keep the existing `get_posterior` methods as is (we can add a default `get_posterior` though as you suggest in `AutoContinuous`).

Reply:
Agreed! We should make separate `get_base_dist` and `get_transform` methods.